src/invariant/LedgerEntryIsValid.cpp

```diff
@@ -174,7 +175,8 @@ LedgerEntryIsValid::checkIsValid(TrustLineEntry const& tl, uint32 version) const
         return fmt::format("TrustLine balance ({}) exceeds limit ({})",
                            tl.balance, tl.limit);
     }
-    if ((tl.flags & ~MASK_TRUSTLINE_FLAGS) != 0)
+    if ((tl.flags & ~MASK_TRUSTLINE_FLAGS) != 0 ||
+        !trustLineFlagIsValid(tl.flags))
```

Comment:
I think `trustLineFlagIsValid` should encompass all validity checks on the flags field, including no extraneous bits set.